pkg/kubectl/cmd/create_clusterrolebinding_test.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"testing"
+
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/rest/fake"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/apis/rbac"
+	cmdtesting "k8s.io/kubernetes/pkg/kubectl/cmd/testing"
+)
+
+var clusterRoleBindingGroupVersion = schema.GroupVersion{Group: "rbac.authorization.k8s.io", Version: "v1alpha1"}
+
+func TestClusterRoleBinding(t *testing.T) {
+	clusterRoleBinding :=&rbac.ClusterRoleBinding{}
+	clusterRoleBinding.Name = "my-clusterrole"
+	f, tf, _, ns := cmdtesting.NewAPIFactory()
+
+        tf.Printer = &testPrinter{}
+	info, _ := runtime.SerializerInfoForMediaType(ns.SupportedMediaTypes(), runtime.ContentTypeJSON)
+	encoder := ns.EncoderForVersion(info.Serializer, clusterRoleBindingGroupVersion)
+
+	tf.Client = &ClusterRoleBindingRESTClient{
+		RESTClient: &fake.RESTClient{
+		APIRegistry:          api.Registry,
+		NegotiatedSerializer: ns,
+		Client: fake.CreateHTTPClient(func(req *http.Request) (*http.Response, error) {
+			responseBinding := &rbac.ClusterRoleBinding{}
+			responseBinding.Name = "fake-binding"
+			return &http.Response{StatusCode: 201, Header: defaultHeader(), Body: ioutil.NopCloser(bytes.NewReader([]byte(runtime.EncodeOrDie(encoder, responseBinding))))}, nil
+		}),
+		},
+	}
+
+	buf := bytes.NewBuffer([]byte{})
+	cmd := NewCmdCreateClusterRoleBinding(f,buf)
+	cmd.Flags().Set("output","name")
+	cmd.Flags().Set("clusterrole","cluster-admin")
+	cmd.Run(cmd,[]string{clusterRoleBinding.Name})
```

Comment:
You should rather run with `--dry-run` and `--output` flags and verify if the created `ClusterRoleBinding` is what you're expecting and not just checking the name. Have a look at https://github.com/kubernetes/kubernetes/blob/f6cb2fce00929f76c0458428c8ddfb533f95fa3f/pkg/kubectl/cmd/create_clusterrole_test.go how the test should look like.